scripts/nix-profile.sh.in

```diff
@@ -1,14 +1,20 @@
-if [ -n "$HOME" ] && [ -n "$USER" ]; then
-    __savedpath="$PATH"
+function configureNix()
+{
+    local __savedpath="$PATH"
     export PATH=@coreutils@
 
     # Set up the per-user profile.
     # This part should be kept in sync with nixpkgs:nixos/modules/programs/shell.nix
 
-    NIX_LINK=$HOME/.nix-profile
-
-    NIX_USER_PROFILE_DIR=@localstatedir@/nix/profiles/per-user/$USER
+    local NIX_LINK=$HOME/.nix-profile
 
+    if [ -z "$NIX_STATE_DIR" ]; then
```

Comment:
One test (nix-profile.sh) expects to be able to override the user profile location.